### ISSUE TITLE
[BUG] Cannot export Gemma/Mistral to ONNX/TensorRT using INT8

### ISSUE THREAD

**Issue by michaelroyzen** [huggingface/optimum#1735]
### System Info

```shell
Optimum main branch, commit bb21ae7f7d572805f6ecdea8e0f02dc6014d57e8
Transformers 4.38.1
OnnxRuntime 1.17.1
PyTorch 2.2.1
TensorRT 8.6.1 (nvcr.io/nvidia/tensorrt:23.10-py3) container
```

### Who can help?

@fxmarty 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

I'm trying to export Gemma/Mistral models from HF to be used in TensorRT in INT8. While the ONNX export completes successfully, I get errors from the TensorRT `trtexec` conversion about symmetric quantization despite enabling those flags in Optimum.

To reproduce, first export a Gemma model to ONNX with static INT8 quantization:

```
from optimum.onnxruntime import AutoQuantizationConfig, ORTModelForFeatureExtraction, AutoCalibrationConfig, ORTQuantizer
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("google/gemma-2b")
onnx_model = ORTModelForFeatureExtraction.from_pretrained("google/gemma-2b", from_transformers=True)
quantizer = ORTQuantizer.from_pretrained(onnx_model)

from functools import partial
import torch

def preprocess_fn(ex, tokenizer):
    encoded_inputs = tokenizer(ex["func_documentation_string"], return_tensors="pt", padding=True)
    position_ids = torch.clamp(torch.cumsum(encoded_inputs['attention_mask'], dim=-1) - 1, min=0)
    # Convert to appropriate tensor type if using PyTorch
    position_ids_tensor = torch.LongTensor(position_ids)
    encoded_inputs['position_ids'] = position_ids_tensor

    return encoded_inputs

calibration_dataset = quantizer.get_calibration_dataset(
    "code_search_net",
    preprocess_function=partial(preprocess_fn, tokenizer=tokenizer),
    num_samples=3,
    dataset_split="train",
)

qconfig = AutoQuantizationConfig.tensorrt(per_channel=False)

calibration_config = AutoCalibrationConfig.minmax(calibration_dataset)
ranges = quantizer.fit(
    dataset=calibration_dataset,
    calibration_config=calibration_config,
    operators_to_quantize=qconfig.operators_to_quantize,
    use_external_data_format=True,
    batch_size=1,
    force_symmetric_range=True,
)

model_quantized_path = quantizer.quantize(
    save_dir="onnx-tensorrt/",
    calibration_tensors_range=ranges,
    quantization_config=qconfig,
    use_external_data_format=True,
)
```

Then, inside the `nvcr.io/nvidia/tensorrt:23.10-py3` container, we build the TRT engine:

```
trtexec \
    --onnx=onnx-tensorrt/model_quantized.onnx \
    --saveEngine=model.plan \
    --minShapes=input_ids:1x512,attention_mask:1x512,position_ids:1x512 \
    --optShapes=input_ids:16x512,attention_mask:16x512,position_ids:16x512 \
    --maxShapes=input_ids:32x512,attention_mask:32x512,position_ids:32x512 \
    --verbose \
    --workspace=768000 \
    --int8
```

which yields the following error:

```
[02/29/2024-08:06:34] [V] [TRT] Parsing node: /layers.0/self_attn/Add_2_output_0_QuantizeLinear [QuantizeLinear]
[02/29/2024-08:06:34] [V] [TRT] Searching for input: /layers.0/self_attn/Add_2_output_0
[02/29/2024-08:06:34] [V] [TRT] Searching for input: /layers.0/self_attn/Add_2_output_0_scale
[02/29/2024-08:06:34] [V] [TRT] Searching for input: /layers.0/self_attn/Add_2_output_0_zero_point
[02/29/2024-08:06:34] [V] [TRT] /layers.0/self_attn/Add_2_output_0_QuantizeLinear [QuantizeLinear] inputs: [/layers.0/self_attn/Add_2_output_0 -> (-1, 8, -1, -1)[FLOAT]], [/layers.0/self_attn/Add_2_output_0_scale -> ()[FLOAT]], [/layers.0/self_attn/Add_2_output_0_zero_point -> ()[INT8]], 
[02/29/2024-08:06:34] [V] [TRT] Registering layer: /layers.0/self_attn/Add_2_output_0_scale for ONNX node: /layers.0/self_attn/Add_2_output_0_scale
[02/29/2024-08:06:34] [E] [TRT] ModelImporter.cpp:771: While parsing node number 2211 [QuantizeLinear -> "/layers.0/self_attn/Add_2_output_0_QuantizeLinear_Output"]:
[02/29/2024-08:06:34] [E] [TRT] ModelImporter.cpp:772: --- Begin node ---
[02/29/2024-08:06:34] [E] [TRT] ModelImporter.cpp:773: input: "/layers.0/self_attn/Add_2_output_0"
input: "/layers.0/self_attn/Add_2_output_0_scale"
input: "/layers.0/self_attn/Add_2_output_0_zero_point"
output: "/layers.0/self_attn/Add_2_output_0_QuantizeLinear_Output"
name: "/layers.0/self_attn/Add_2_output_0_QuantizeLinear"
op_type: "QuantizeLinear"

[02/29/2024-08:06:34] [E] [TRT] ModelImporter.cpp:774: --- End node ---
[02/29/2024-08:06:34] [E] [TRT] ModelImporter.cpp:777: ERROR: builtin_op_importers.cpp:1221 In function QuantDequantLinearHelper:
[6] Assertion failed: shiftIsAllZeros(zeroPoint) && "TensorRT only supports symmetric quantization. The zero point for the QuantizeLinear/DequantizeLinear operator must be all zeros."
[02/29/2024-08:06:34] [E] Failed to parse onnx file
[02/29/2024-08:06:35] [I] Finished parsing network model. Parse time: 15.6036
[02/29/2024-08:06:35] [E] Parsing model failed
[02/29/2024-08:06:35] [E] Failed to create engine from model or file.
[02/29/2024-08:06:35] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8601] # trtexec --onnx=onnx-tensorrt/model_quantized.onnx --saveEngine=model.plan --minShapes=input_ids:1x512,attention_mask:1x512,position_ids:1x512 --optShapes=input_ids:16x512,attention_mask:16x512,position_ids:16x512 --maxShapes=input_ids:32x512,attention_mask:32x512,position_ids:32x512 --verbose --workspace=768000 --int8
```

It seems like the weights aren't symmetric after all. Would appreciate your help @fxmarty! Thanks in advance.

### Expected behavior

The weights are symmetric and the TensorRT engine successfully builds with INT8.

**Comment by michaelroyzen** [comment]
Despite using a TensorRT config and `force_symmetric_range=True`, it seems the weights are still asymmetrical @fxmarty.

**Comment by fxmarty** [comment]
@michaelroyzen I can reproduce the issue. Could you inspect your model with netron to inspect the node `/layers.0/self_attn/Add_2_output_0_QuantizeLinear`?

Trying on a dummy model (https://huggingface.co/fxmarty/tiny-random-GemmaForCausalLM), it appears that at least one node is not quantized properly, but others are fine: 
![image](https://github.com/huggingface/optimum/assets/9808326/481fc197-fbde-41fe-9eaf-56590628a3e0)

I'm not sure whether this is due to using a dummy model, a very small dataset or a bug in ORT.

**Comment by fxmarty** [comment]
Can you try:

```python
qconfig.operators_to_quantize = ['Conv', 'ConvTranspose', 'Gemm', 'Clip', 'Relu', 'Reshape', 'Transpose', 'Squeeze', 'Unsqueeze', 'Resize', 'MaxPool', 'AveragePool', 'MatMul', 'Split', 'Gather', 'Where', 'InstanceNormalization', 'LayerNormalization']
```

(or a subset of that - simply removing `"Softmax"`)

Also maybe useful to you: https://github.com/huggingface/optimum/blob/bb21ae7f7d572805f6ecdea8e0f02dc6014d57e8/examples/onnxruntime/quantization/text-classification/run_glue.py#L467-L476

After that all quantization nodes are symmetric, but you may get an other error:

```
[02/29/2024-09:52:02] [V] [TRT] logits_DequantizeLinear [DequantizeLinear] inputs: [logits_QuantizeLinear_Output -> (-1, -1, 256000)[FLOAT]], [logits_scale -> ()[FLOAT]], [logits_zero_point -> ()[INT8]], 
[02/29/2024-09:52:02] [V] [TRT] Registering tensor: logits_349 for ONNX tensor: logits
[02/29/2024-09:52:02] [V] [TRT] logits_DequantizeLinear [DequantizeLinear] outputs: [logits -> (-1, -1, 256000)[FLOAT]], 
[02/29/2024-09:52:02] [V] [TRT] Marking logits_349 as output: logits
[02/29/2024-09:52:02] [I] Finished parsing network model. Parse time: 0.0642945
[02/29/2024-09:52:02] [I] FP32 and INT8 precisions have been specified - more performance might be enabled by additionally specifying --fp16 or --best
[02/29/2024-09:52:02] [W] [TRT] Calibrator won't be used in explicit precision mode. Use quantization aware training to generate network with Quantize/Dequantize nodes.
[02/29/2024-09:52:02] [E] Error[4]: [graphShapeAnalyzer.cpp::analyzeShapes::2013] Error Code 4: Miscellaneous (IElementWiseLayer /model/And: broadcast dimensions must be conformable)
[02/29/2024-09:52:02] [E] Engine could not be created from network
[02/29/2024-09:52:02] [E] Building engine failed
[02/29/2024-09:52:02] [E] Failed to create engine from model or file.
[02/29/2024-09:52:02] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8601] # trtexec --onnx=model_quantized.onnx --saveEngine=model.plan --minShapes=input_ids:1x512,attention_mask:1x512,position_ids:1x512 --optShapes=input_ids:16x512,attention_mask:16x512,position_ids:16x512 --maxShapes=input_ids:32x512,attention_mask:32x512,position_ids:32x512 --verbose --int8

```

**Comment by fxmarty** [comment]
Same error on `nvcr.io/nvidia/tensorrt:24.01-py3`. Sounds like a bug in TRT.

**Comment by michaelroyzen** [comment]
We do use TRT-LLM for generation, but our goal here is to use Mistral/Gemma for embeddings. TRT-LLM only returns logits, not the last hidden states pre-projection.

**Comment by IlyasMoutawwakil** [comment]
@michaelroyzen were you able to confirm that the issue is fixed ? as reported in https://github.com/NVIDIA/TensorRT/issues/3688

**Comment by michaelroyzen** [comment]
Can't confirm yet as TensorRT 10.0 has not been released yet. @IlyasMoutawwakil

**Comment by fxmarty** [comment]
Closing as it is a TRT bug.

**Comment by CHNtentes** [comment]
@michaelroyzen Hi. Is your static quantized model working normally? I tried quantizing MiniCPM model using similar approach, however it just generate some garbage.

**Comment by geraldstanje** [comment]
hi @michaelroyzen why do you use ORTQuantizer? cant you just quantize the model using trtexec?
